.github/ci.yml

```diff
@@ -0,0 +1,28 @@
+on: [push, pull_request]
+name: CI
+jobs:
+  test:
```

Comment:
Let's have separate jobs. Instead of a catch-all `test`, let's have  `lint`, `format`, and `unit-tests` jobs (this means setting up the deps on all). This will make it easier to know why a PR is failing as they run in parallel.